crates/cache/src/lib.rs

```diff
@@ -44,6 +45,24 @@ pub enum Error {
 
 pub type Result<T, E = Error> = std::result::Result<T, E>;
 
+pub enum ResultCacheStatus {
+    Cached,
+    NotCached,
+    CacheDisabled,
+}
```

Comment:
```suggestion
pub enum RetrievalResult {
    Hit,
    Miss
}
```

Don't put disabled in here.

Reply:
Like the proposed approach with Option<RetrievalResult>. Updated